### PR TITLE
Add a separate option to control solution crawler's support of generators

### PR DESCRIPTION
**Pull request by jasonmalinowski**
When the solution crawler was originally given support for analyzing source generated files, it was put under the same user-visible option and feature flag as our support for connecting opened source generated files to the workspace. This was done simply out convenience: why have two checkboxes if one will do? Unfortunately, once we went to turn on this checkbox by default, we ran into performance issues with the solution crawler, even though functionally the first part (enabling features in open source generated files) was still fine. Since we're also now actively getting rid of the solution crawler, it doesn't make any sense for us to fix those performance issues any further, but in the mean time we're still blocked on enabling the functionality we want.

This PR simply breaks out the solution crawler support for source generated files into a separate option, which we can then enable once it's safe to do so without holding up the open-file feature support. Hopefully, the solution crawler will be deprecated soon enough and this can all be deleted. But, just in case we run into issues, we have a backup plan without holding up feature work that's been ready to go.

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
FYI to @sharwell.

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingWorkCoordinator.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ internal sealed partial class UnitTestingWorkCoordinator : IUnitTestingWorkCoord
 
             private readonly CountLogAggregator<WorkspaceChangeKind> _logAggregator = new();
             private readonly IAsynchronousOperationListener _listener;
-            private readonly IWorkspaceConfigurationService? _workspaceConfigurationService;
+            private readonly Microsoft.CodeAnalysis.SolutionCrawler.ISolutionCrawlerOptionsService? _solutionCrawlerOptionsService;
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingWorkCoordinator.cs:34]: Note the unit testing fork of the solution crawler was still using the same underlying option as the mainline solution crawler; this maintains that.

File: src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerOptionsService.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.SolutionCrawler
+{
+    internal interface ISolutionCrawlerOptionsService : IWorkspaceService
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerOptionsService.cs:12]: I introduced a new service as opposed to adding a new option on the existing services, so there was no risk this doesn't get deleted once we delete the solution crawler. Yes, I'm optimizing for deleting of this over anything else. :smile:

File: src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs [hunk 3]
```diff
@@ -47,7 +47,7 @@ internal sealed partial class WorkCoordinator
 
                 _listener = listener;
                 _documentTrackingService = _registration.Workspace.Services.GetRequiredService<IDocumentTrackingService>();
-                _workspaceConfigurationService = _registration.Workspace.Services.GetService<IWorkspaceConfigurationService>();
+                _solutionCrawlerOptions = _registration.Workspace.Services.GetService<ISolutionCrawlerOptionsService>();
```
Comment (genlu) [inline src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:50]: Why not have a default implementation at this layer instead of using a nullbale?

**Review by genlu — APPROVED** [review]

File: src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs (hunk 3 above)
Comment (jasonmalinowski) [reply src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:50]: @genlu Preserving the existing behavior; I don't know why that would have been done with the other service either, but I don't want to find out.

**Pull request merged**